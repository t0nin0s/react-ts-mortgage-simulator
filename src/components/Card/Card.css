:local(.card) {
  /*font-size: 1rem;*/
  margin: 0 auto;
  width: 90%;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  webkit-border-radius: 0.6em;
  -moz-border-radius: 0.6em;
  border-radius: 0.6em;
  -webkit-box-shadow: 5px 5px 15px 0px rgba(136,136,136,1);
  -moz-box-shadow: 5px 5px 15px 0px rgba(136,136,136,1);
  box-shadow: 5px 5px 15px 0px rgba(136,136,136,1);
  overflow:hidden;
}

:local(.card__title) {
  background-color: #A06FEF;
  color: white;
  text-align: center;
  padding: 1rem;
}

:local(.card__icon) {
  text-align: center;
  padding-top: 4rem;
  padding-bottom: 1.5rem;
}

:local(.card__options) {
  height: 35rem;
}

:local(.card__footer) {
  text-align: center;
  text-decoration: underline;
  color: #A0A0A0;
  border-top: 1px solid #ECECEC;
  padding: 1.5rem 0;
  min-height: 5rem;
}


@media all and (min-width: 460px) {
  :local(.card) {
    max-width: 40rem;
  }
}
