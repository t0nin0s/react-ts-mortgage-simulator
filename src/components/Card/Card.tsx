import * as React from 'react';
import * as FontAwesome from 'react-fontawesome';
const styles = require('./Card.css');

interface CardProps {
  formTitle: string;
  icon: string;
  footer?: boolean;
}

const Card: React.SFC<CardProps> = (props) => {
  return (
    <div className={styles.card}>
      <div className={styles.card__title}>
        {props.formTitle}
      </div>
      <div className={styles.card__icon}>
        <FontAwesome
          name={props.icon}
          size="4x"
          style={{ color: '#A06FEF' }}
        />
      </div>
      <div className={styles.card__options}>
        {props.children}
      </div>
      <div className={styles.card__footer}>
        { props.footer &&
          'Why do you need this?'
        }
      </div>
    </div>
  );
};

export default Card;
