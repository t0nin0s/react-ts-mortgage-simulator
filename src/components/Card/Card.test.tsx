import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Card from './Card'

describe('Card renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <Card
        formTitle=''
        icon=''
        footer />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
