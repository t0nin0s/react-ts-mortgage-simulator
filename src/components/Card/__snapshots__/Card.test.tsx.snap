// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders as expected should render as expected 1`] = `
<div>
  <div />
  <div>
    <FontAwesome
      name=""
      size="4x"
      style={
        Object {
          "color": "#A06FEF",
        }
      }
    />
  </div>
  <div />
  <div>
    Why do you need this?
  </div>
</div>
`;
